system-addon/lib/TopSitesFeed.jsm

```diff
@@ -52,7 +52,12 @@ this.TopSitesFeed = class TopSitesFeed {
   }
   async getLinksWithDefaults(action) {
     let frecent = await NewTabUtils.activityStreamLinks.getTopSites();
-    const notBlockedDefaultSites = DEFAULT_TOP_SITES.filter(site => !NewTabUtils.blockedLinks.isBlocked({url: site.url}));
+    const notBlockedDefaultSites = DEFAULT_TOP_SITES
+      .filter(site => !NewTabUtils.blockedLinks.isBlocked({url: site.url}))
+      .map(site => {
+        const tippySite = this._tippyTopProvider.processSite(site);
```

Comment:
@rlr should tippy top only be applying to the default sites and not other links? If so, should this `processSite` in `getLinksWithDefaults` remove the need to call `processSite` from `refresh` ?

Reply:
@Mardak We want to run tippy top on all sites. Although, for now (in 56), we only have assets for default sites. I still think it's a better experience if you get the nicer screenshot for facebook, for example, whether you already visited it or it's displaying because of it's default status.

Reply:
Adding this dependency on tippy top doesn't seem right when metadata might be replacing it. @piatra any reason you can't just use `eTLD` with `shortURL`?

Reply:
DEFAULT_URLS and frecent website information is never merged. You will have two `amazon` entities (the default and the actual visit) and the default_url entity will never receive eTLD information. It also won't be discarded by dedupe because we won't be able to extract the correct hostname key.

Reply:
What do you mean default_urls and frecent never get merged? Isn't the whole point of this method to combine the frecent links with defaults?

Reply:
https://github.com/mozilla/activity-stream/blob/560e9b62a2bde24feb795f16418ffd49df54ee95/system-addon/lib/TopSitesFeed.jsm#L77 We dedupe pinned and default topsites. If an item is found in both groups, both will show up in your topsites. It will not be deduped because eTLD field will never be present for default topsites.

Reply:
Right... so why can't we add in eTLD as I suggested?